test/system/groups/samples_test.rb

```diff
@@ -95,22 +104,22 @@ def setup
     end
 
     test 'can sort the list of samples' do
-      freeze_time
       visit group_samples_url(@group)
 
+      # Because PUIDs are not always generated the same, issues regarding order have occurred when hard testing
+      # the expected ordering of samples based on PUID. To resolve this, we will gather the first 4 PUIDs and ensure
+      # they are ordered as expected against one another.
       assert_selector 'table tbody#group-samples-table-body tr', count: 20
-      within first('table tbody#group-samples-table-body') do
-        assert_selector 'tr:first-child td:first-child', text: @sample1.puid
-        assert_selector 'tr:first-child td:nth-child(2)', text: @sample1.name
+      puids = retrieve_puids
+      3.times do |n|
```

Comment:
**nitpick:** Instead of hard coding the 3, let do `(puids.length -1)`

Reply:
Changed in 0cd4235004b2869291f8b90952bf976b2c908388

Reply:
Awesome, thanks